### PR TITLE
Install root package.json modules before demeteorizing

### PR DESCRIPTION
**Pull request by theworkflow**
- Install root `package.json` modules
- Copy over `node_modules` directory to `$OUTPUT_DIR`
- [x] Build image
- [x] Test on staging

### CODE REVIEW

**Comment by franvarney** [comment]
👍

**Comment by theworkflow** [comment]
The `meteor build` command is sometimes inconsistent. It will sometimes work and sometimes not work with the exact same bundle.

**Comment by tzmanics** [comment]
:+1:

**Comment by theworkflow** [comment]
Image built and tested on staging successfully.

File: bin/build [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
 set -e
 
+# Uncomment the following line while debugging
```
Comment (jackboberg) [inline bin/build]: I think we can just add the `x` to the `set` command above when debugging, no need for commented code

File: bootstrap.sh [hunk 2]
```diff
@@ -39,9 +41,10 @@ cd /opt
 # Install nvm
 mkdir -p $NVM_DIR
 curl https://raw.githubusercontent.com/creationix/nvm/v0.31.0/install.sh | bash
+source $PROFILE
 
 # Ensure mop can use nvm, but not write to it
-chown mop:mop /opt/nvm/nvm.sh
+chown -R mop:mop /opt/nvm/nvm.sh
```
Comment (jackboberg) [inline bootstrap.sh]: why `-R` when targeting a file?

File: bin/build [hunk 3]
```diff
@@ -28,9 +44,20 @@ cp -R /opt/home/.meteor $HOME
 # --json || --debug
 
 cd $INPUT_DIR
-printf "Using $(meteor --version).\n"
+METEOR_VERSION=$(cat $HOME/.meteor/version)
```
Comment (jackboberg) [inline bin/build:45]: This is not the same `$HOME` as in [bootstrap.sh:67](https://github.com/onmodulus/docker-build-meteor/pull/12/files#diff-3af436e6760a0c5f1e3d16a3384853a8R67)

Is that an issue?

Comment (theworkflow) [inline bin/build:45]: No issue. The meteor distribution is copied over to the `mop` user's `$HOME/.meteor` directory.
In [bootstrap](https://github.com/onmodulus/docker-build-meteor/pull/12/files#diff-3af436e6760a0c5f1e3d16a3384853a8R64) I created a file called `version` inside the meteor distribution.

File: bin/build [hunk 4]
```diff
@@ -41,44 +66,15 @@ findFile() {
 }
 
 # Recursively search input directory for a package.json
-PACKAGE_PATH=$(findFile "package.json" $OUTPUT_DIR)
+PACKAGE_PATH=$(findFile "package.json" $OUTPUT_DIR/bundle)
 
 if [[ $PACKAGE_PATH ]]; then
   printf "Package found: ${PACKAGE_PATH/$OUTPUT_DIR/}\n"
```
Comment (jackboberg) [inline bin/build]: This ends up printing kind of redundantly:

```
Package found: /bundle/programs/server/package.json
Installing packages from /mnt/output/bundle/programs/server/package.json
```

**Comment by jackboberg** [comment]
My tests on staging seem to have resolved outstanding issues. :shipit:

File: bin/build (hunk 4 above)
Comment (theworkflow) [inline bin/build]: I agree. I could remove that `printf` line

**Comment by theworkflow** [comment]
@jackboberg whatever outstanding issues this PR addresses can you note them here please?

**Pull request merged**